### PR TITLE
feat: Adding Default decorator for non-existing key value pairs on .env file

### LINKED ISSUE #12: [Feature Request]Can optional parameters be supported?
Can **envied** support optional parameters when we use `obfuscate` ? Like `maybeGet` in [dotEnv](https://pub.dev/packages/flutter_dotenv)
Or Could we assign a value using `@Default` decorator?

### PR DESCRIPTION
**Pull request by kaankoken**
- Add `Default` decorator to `envied_base.dart`
- Remove the unnecessary package import, `dotEnv`, is removed
- Add `Parser` function is taken from [dotEnv](https://pub.dev/packages/flutter_dotenv)
- Update `Default` decorator check is added to `generator.dart`
- Add tests for `generator_tests.dart` & `envy_test.dart` is added
- Add examples under `examples` updated using the `Default` decorator
- Update temporarily packages directed to the forked repository(commits will be reverted)
- Readme updated with using Default decorator
- Resolves #12

### CODE REVIEW

**Review by petercinibulk — CHANGES_REQUESTED** [review]
Why did you make changes tp the pubspec?
Specifically added publish_to: none and making it poing to your forked version of envied?

Please fix those and revert them back.

**Comment by kaankoken** [comment]
> Why did you make changes to the pubspec? Specifically added publish_to: none and making it point to your forked version of envied?
> 
> Please fix those and revert them back.

I actually added `publish_to: none` to use in the example & in my personal project. Otherwise, I did not let to use it the projects.

I will revert it.

**Comment by kaankoken** [comment]
@petercinibulk You can check it

**Comment by ThexXTURBOXx** [comment]
I am unsure if adding another annotation would be the way to go here.
Maybe another parameter inside the current `@EnviedField` might be preferable

**Comment by petercinibulk** [comment]
@ThexXTURBOXx I agree with you. I'd like to keep the annotations to a minimum as a good practice

File: packages/envied/README.md
Comment (ThexXTURBOXx) [inline packages/envied/README.md:125]: I think, this still needs to be changed. Apart from that, I have not tested the new code yet, but it looks fine now :)

**Review by ThexXTURBOXx — CHANGES_REQUESTED** [review]

File: packages/envied/README.md
Comment (kaankoken) [reply packages/envied/README.md:125]: I will push the new changes as soon as I tested. I am testing on a different branch.

Comment (kaankoken) [reply packages/envied/README.md:125]: @ThexXTURBOXx @petercinibulk @techouse @daiki1003 if you would like to check it out  you can check over [here](https://github.com/kaankoken/envied/tree/feature/test-default-value).

File: packages/envied_generator/lib/src/load_envs.dart
Comment (ThexXTURBOXx) [inline packages/envied_generator/lib/src/load_envs.dart:4]: That `ignore` seems to be superfluous

**Review by ThexXTURBOXx — COMMENTED** [review]
Just another tiny thing, but now at least I am satisfied :)

File: packages/envied_generator/lib/src/load_envs.dart
Comment (kaankoken) [reply packages/envied_generator/lib/src/load_envs.dart:4]: I could remove it. No worries.

**Review by ThexXTURBOXx — APPROVED** [review]

**Comment by kaankoken** [comment]
the `ci` is not passing since the `defaultValue` is not in version `0.2.4`.

**Comment by kaankoken** [comment]
image

Since `defaultValue` is not version `0.2.4`, `dart analyze` crates these outputs

**Comment by ThexXTURBOXx** [comment]
@kaankoken I think, you can ignore these errors. They are normal in multi-project repos like this one :)

**Comment by kaankoken** [comment]
> @kaankoken I think, you can ignore these errors. They are normal in multi-project repos like this one :)

Okay, thanks. I am waiting for approval.

**Pull request merged**
